### PR TITLE
add libc property

### PR DESCRIPTION
**Pull request by meik99**

### CODE REVIEW

File: deploy/cr-apm.yaml [hunk 1]
```diff
@@ -49,3 +50,8 @@ spec:
   #   limits:
   #     cpu: 500m
   #     memory: 100M
+
+  # Optional: define which core library for C should be used.
```
Comment (baichinger) [inline deploy/cr-apm.yaml:54]: Could be rephrased to "defines the C standard library used."

File: pkg/apis/dynatrace/v1alpha1/oneagentapm_types.go [hunk 2]
```diff
@@ -45,6 +45,13 @@ type OneAgentAPMSpec struct {
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Resource Requirements"
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:advanced,urn:alm:descriptor:com.tectonic.ui:resourceRequirements"
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Optional: define which core library for C should be used.
+	// Can be set to "musl" to use musl instead of glibc
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="C Core Library"
```
Comment (baichinger) [inline pkg/apis/dynatrace/v1alpha1/oneagentapm_types.go:52]: Please use _C standard library_ as well.

File: pkg/webhook/server/flavor.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	"github.com/Dynatrace/dynatrace-oneagent-operator/pkg/controller/utils"
+	dtwebhook "github.com/Dynatrace/dynatrace-oneagent-operator/pkg/webhook"
+)
+
+func getFlavor(libc string /*oa.Spec.LibC*/, annotations map[string]string /* pod.Annotations */) string {
+	flavor := url.QueryEscape(utils.GetField(annotations, dtwebhook.AnnotationFlavor, libc))
+	if flavor != "musl" {
+		return "default"
+	}
+	return flavor
```
Comment (baichinger) [inline pkg/webhook/server/flavor.go:15]:
```suggestion
	if flavor = "musl" {
		return flavor
	}
	return "default
```

**Review by baichinger — COMMENTED** [review]
Will this cause confusion when the user needs to know that annotation _flavor_ is actually the same as custom resource field _libc_. It will get even more strange when values of them do not match: `musl` vs. `default` vs. `glibc`.

Therefore, shouldn't we simply stick with the term _flavor_?

**Review by lrgar — APPROVED** [review]

**Review by DTMad — APPROVED** [review]

**Pull request merged**